packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -121,8 +127,8 @@ export function buildCategoryFacet(
         })
       ),
 
-    /** Sorts the category facet values according to the passed criterion.
-     * @param {CategoryFacetSortCriterion} criterion The criterion to sort values by.
+    /** Sorts the facet values according to the specified criterion.
+     * @param criterion (CategoryFacetSortCriterion) The criterion to sort values by.
```

Comment:
Similar to this comment: 

https://github.com/coveo/ui-kit/pull/370#discussion_r544418144

Reply:
IIRC, someone from the docs team decided way back that this was the way were going to annotate using typedoc, and so in terms of generation for the docs site, changing the syntax here would require some changes on our end as well. If doing it this way causes a problem with intellisense, it's worth the change, but otherwise I think we should document the way we've been doing so far.

Reply:
🤔  I'm not sure I understand how this affect the doc website generation.

There's a bug in Typedoc so that it does not use the correct format ? 

Because Typedoc, normally, is just "standard" jsdoc with inferred types from TypeScript.

Reply:
We don't use typedoc to actually generate the doc site reference pages; we instead use typedoc to generate a doc.json file, and we then have a script we've written that extracts all the information from that file and creates pages on our site. So right now, we can just take the whole comment, including the type, and pull it into the right spot. I'm sure it's relatively straightforward to change- I'll ask the docs team how much work it would be on our end

Reply:
Upon investigation, it seems that we started doing this with actions, because the types do not appear in the docgen at all, so in order to get them we added them in directly to the comment. Since typedoc can infer the types for controller functions, I don't think we even need to include the type there at all, at least for the doc site generation. Does that sound right to you?

Reply:
Ah I see, in this case we might not need it for the website generation.

But this jsdoc will still appear for any end users in their IDE (say, vscode). 

So, using proper jsdoc format, with the types, still has a benefit of displaying correctly/nicely for consumers of the library in that case.

Reply:
Check out #376 to see what I did there- I didn't include any types and intellisense still picked them up correctly. If that's how we want to proceed, I'll make those changes here. We'll also have to go fix the controller docs PRs that have already been merged (basic facet + resultlist + anything else you've merged)